tests/tests_fabric/plugins/precision/test_bitsandbytes.py

```diff
@@ -107,33 +110,79 @@ def __init__(self):
             self.ln = torch.nn.LayerNorm(2)
 
     state_dict = MyModel().state_dict()
-
     fabric = Fabric(devices=1, plugins=BitsandbytesPrecision(*args))
     with fabric.init_module():
         model = MyModel()
+
     # the model was instantiated on-device and quantized straight away
     assert model.l.weight.device.type == "cuda"
     assert model.l.weight.dtype == expected
     # this has no impact
     model = fabric.setup(model)
     assert model.l.weight.device.type == "cuda"
     assert model.l.weight.dtype == expected
-    # state dict loading still works even thought the weights are quantized
+    # unquantized state dict loading still works even thought the weights are quantized
     weight_before = model.l.weight.data.clone()
     keys = model.load_state_dict(state_dict, strict=True)
     assert not keys.missing_keys
     assert not torch.equal(weight_before, model.l.weight.data)
     assert model.l.weight.device.type == "cuda"
     assert model.l.weight.dtype == expected
+    # FIXME: test quantized state dict weights can be loaded back
+    # FIXME: test state_dict can be unquantized and can be loaded into the original model
```

Comment:
Something like this, yes.
But this is possible thanks to Fabric, as I see.
I'll add in Lit-GPT a function to dequantize weights for the cases when a user wants to use the model after quantization/dequantization elsewhere.
Or you are planning to add this soon?
https://github.com/Lightning-AI/pytorch-lightning/blob/dafe4e399213ee386fc504edaa7e9d82afe6ecae/tests/tests_fabric/plugins/precision/test_bitsandbytes.py#L135

Reply:
Not planning to work on this. I tagged you here because I think it would be best that you add dequantization support to Fabric, then lit-gpt can call these functions

Reply:
... and add tests.
Now I get it.
Ok, I'll make a PR.